tests/utils/test_hasher.py

```diff
@@ -0,0 +1,22 @@
+from hypothesis import given
+from hypothesis.strategies import emails
+
+from app.utils.hasher import Hasher
+
+
+class TestHasher:
+    def test_hasher_using_argon2id_with_expected_parameters(self):
+        hasher = Hasher()
+
+        hash = hasher.hash('abc123')
+
+        assert hash.startswith('$argon2id$')
+        assert '$m=15360,t=2,p=1$' in hash
+
+    @given(emails())
+    def test_hash_verifies_correctly(self, value):
```

Comment:
So that uses a precompiled collection of emails to test that the hash verifies correctly? Nifty :3

Reply:
Not quite - `hypothesis` generates random (different) emails each test run. So it's possible that this test could become flaky if there is some value that eventually doesn't work - but at least we'll notice at some point. But effectively yes - this generates a load of different valid email addresses and checks that they all work. :)

Reply:
Ahh, ok, quite interesting, thanks a lot!